site/app/models/RainbowCustomization.php

```diff
@@ -17,18 +17,35 @@
 class RainbowCustomization extends AbstractModel {
     /**/
     protected $core;
-    private $bucket_counts = [];               // Keep track of how many items are in each bucket
     /**
-     * @var int[]
+     * @var array<string,int>
      */
-    private array $bucket_remove_lowest = [];               // get how many droplowest are in each bucket
-    private $customization_data = [];
-    private $has_error;
-    private $error_messages;
-    private $used_buckets = [];
-    private $available_buckets;
-    private $RCJSON;                           // This is the customization.json php object, or null if it wasn't found
-    private $sections;                         // Contains section ids mapped to labels
+    private array $bucket_counts = [];                  // Keep track of how many items are in each bucket
+    /**
+     * @var array<string,int>
+     */
+    private array $bucket_remove_lowest = [];           // get how many droplowest are in each bucket
+    /**
+     * @var array<string,array<mixed>>
+     */
+    private array $customization_data = [];
+    /**
+     * @var string
+     */
+    private string $has_error;
```

Comment:
This may be outside the scope of this PR, but is there a reason that $has_error is a string?

Reply:
Not a good reason. It uses "false" as a value.

Reply:
It would be good to leave a TODO comment with suggestions for how to make it better (i.e., change to a boolean), or change the type in this PR if it's not too difficult.